DllImportGenerator/DllImportGenerator/Marshalling/MarshallingGenerator.cs

```diff
@@ -263,5 +275,64 @@ private static IMarshallingGenerator CreateStringMarshaller(TypePositionInfo inf
 
             throw new MarshallingNotSupportedException(info, context);
         }
+
+        
+        private static ExpressionSyntax GetNumElementsExpressionFromMarshallingInfo(TypePositionInfo info, StubCodeContext context)
+        {
+            ExpressionSyntax numElementsExpression;
+            if (info.MarshallingAttributeInfo is MarshalAsInfo { ArraySizeParamIndex: short index, ArraySizeConst: int constSize })
```

Comment:
I really dislike this deeply nested local declarations. They aren't clearer to me but rather seem to be creating side-effects. Let's just make them local when we end the if branch.

Reply:
I personally think this way is cleaner, but I can rearrange things and gather these properties in a more traditional C# style.